src/ControlSurface/ScoreControlSurface.qml

```diff
@@ -31,26 +31,102 @@ import Variable.Global 1.0
 
 Column {
     id: controlSurfaceListColumn
-    spacing: 5
+
     property string name
 
-    Rectangle {
-        width: parent.width
-        height: 5
-        color: Skin.gray1
+    // Receiving information about controls in a control surface from score
+    Connections {
+        target: controlSurface
+
+        // Adding controls in a control surface
+        function onAppendControls(m) {
+            controlSurfaceName.text = m.Name
+            var i = 0
+            var controlMessage = m.Controls[i]
+            while (controlMessage) {
+                switch (controlMessage.uuid) {
+                case Uuid.floatSliderUUID:
+                    // Float Slider
+                    scoreSliders.appendSlider(controlMessage, i)
+                    break
+
+                case Uuid.logFloatSliderUUID:
+                    // Log Float Slider
+                    scoreSliders.appendSlider(controlMessage, i)
+                    break
+
+                case Uuid.intSliderUUID:
+                    // Int Slider
+                    scoreSliders.appendSlider(controlMessage, i)
+                    break
+
+                case Uuid.impulseButtonUUID:
+                    // Impulse Button
+                    scoreImpulseButtons.appendImpulseButton(controlMessage)
+                    break
+
+                case Uuid.buttonUUID:
+                    // Button
+                    scoreButtons.appendButton(controlMessage)
+                    break
+
+                case Uuid.colorPickerUUID:
+                    // Colorpicker
+                    scoreColorpicker.visible = true
+                    scoreColorpicker.appendColorpicker(controlMessage)
+                    break
+
+                case Uuid.positionUUID:
+                    // Position
+                    scorePosition.visible = true
+                    scorePosition.appendPosition(controlMessage)
+                    break
+
+                case Uuid.comboBoxUUID:
+                    // ComboBox
+                    scoreComboBox.appendComboBox(controlMessage)
+                    break
+
+                default:
+                    break;
+                }
+                i++
+                controlMessage = m.Controls[i]
+            }
+        }
+
+        // Modifying controls in a control surface
+        function onModifyControl(m) {
+
+            if (m.Message === "ControlSurfaceControl") {
+                scoreSliders.modifySlider(m)
```

Comment:
an idea: what about instead calling the method "modify" for these four things and having references to these four objects in a single scoreControls container ? so that one could do

     for(let controls of scoreControls) {
       controls.modify();
     }

or

     scoreControls.forEach(controls => controls.modify());